### ISSUE TITLE
Strange wrapping on heavily indented text

### ISSUE THREAD

**Issue by alexarice** [WGUNDERWOOD/tex-fmt#6]
The following (with the text being all on the same line):
```
\begin{definition}
  \begin{definition}
    \begin{definition}
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    \end{definition}
  \end{definition}
\end{definition}
```
is formatted to:
```
\begin{definition}
  \begin{definition}
    \begin{definition}
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
      veniam,
      quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse
      cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
      non
      proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    \end{definition}
  \end{definition}
\end{definition}
```
I think the formatter is doing a wrapping pass followed by an indentation pass followed by another wrapping pass?

**Comment by WGUNDERWOOD** [comment]
Yes, this is expected behaviour at the moment, as the formatter alternates between wrapping and indenting as you suggest. I am hoping to restructure this to remove the multiple passes framework at some point.

**Comment by alexarice** [comment]
A potential band-aid fix to this could be for line wrapping to not count the initial indent as characters, this would mean that the text starts going past 80 characters but the line width should be bounded by your indent level

**Comment by WGUNDERWOOD** [comment]
Yes this is possible but I would really like the formatter to adhere to 80 characters whenever it can. I'll think about this more.

**Comment by WGUNDERWOOD** [comment]
I am working on a solution as part of a refactor in the `one-pass` branch, which does the following:
- If a line is less than `WRAP_MAX = 80` characters, it is kept.
- Otherwise, it is wrapped to be at most `WRAP_MIN = 70` characters.
This way, after the line is re-indented, it will most likely not need wrapping again.

**Comment by WGUNDERWOOD** [comment]
Implemented in 958940507651a61fa4b3c9fa92e9fea8fe577509. Let me know if this is still not desirable and I can reopen the issue.
